### ISSUE TITLE
GNU Make 4.1+dbg1.1: run quit with error and can't quit  in nested depth

### ISSUE THREAD

**Issue by liugang** [rocky/remake#24]
GNU Make 4.1+dbg1.1: run quit with error and can't quit  in nested depth,
GNU Make 3.82+dbg0.9 works fine.

# cat Makefile
```
all: foo

foo:
	$(MAKE) bar

bar:
	@echo "This is bar recipe"
```

# remake -X
```
GNU Make 4.1+dbg1.1
Built for x86_64-unknown-linux-gnu
Copyright (C) 1988-2014 Free Software Foundation, Inc.
Copyright (C) 2015 Rocky Bernstein.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
Updating goal targets....
-> (/home/galen_liu/remake_test/Makefile:3)
Will update target:foo: 
remake<0> n
   File `foo' does not exist.
-- (/home/galen_liu/remake_test/Makefile:3)
Will update target:foo: 
remake<1> n
  Must remake target `foo'.
Makefile:4: target 'foo' does not exist
##>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
remake bar
##<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Reading makefiles...
Updating makefiles....
Updating goal targets....
-> (/home/galen_liu/remake_test/Makefile:6)
Will update target:bar: 
remake<<0>> q
remake: That's all, folks...
Makefile:4: recipe for target 'foo' failed
Makefile:3: *** [foo] Error 77

#0  foo at /home/galen_liu/remake_test/Makefile:3
#1  all at /home/galen_liu/remake_test/Makefile:1

***Entering debugger because we encountered a fatal error.
** Exiting the debugger will exit make with exit code 77.
!! (/home/galen_liu/remake_test/Makefile:3)
Will update target:foo
remake<2> q
remake: That's all, folks...
remake: *** Waiting for unfinished jobs....
remake: *** wait: No child processes.  Stop.
```

**Comment by rocky** [comment]
Should be fixed in commit 1396796

**Comment by liugang** [comment]
>  Should be fixed in commit 1396796

Thanks, it fixed this issue.
